pkg/cloud/services/eks/cluster.go

```diff
@@ -0,0 +1,502 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eks
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/eks"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/version"
+
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/awserrors"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/wait"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/internal/tristate"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/record"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-aws/api/v1alpha3"
+	infrav1exp "sigs.k8s.io/cluster-api-provider-aws/exp/api/v1alpha3"
+)
+
+func (s *Service) reconcileCluster(ctx context.Context) error {
+	s.scope.V(2).Info("Reconciling EKS cluster")
+
+	cluster, err := s.describeEKSCluster()
+	if err != nil {
+		return errors.Wrap(err, "failed to describe eks clusters")
+	}
+
+	if cluster == nil {
+		cluster, err = s.createCluster()
+		if err != nil {
+			return errors.Wrap(err, "failed to create cluster")
+		}
+		s.scope.Info("Created EKS control plane: %s", *cluster.Name)
```

Comment:
these loggers are ("message", ...keysAndValues)

so this prints `"msg"="Created EKS control plane: %s"` and then has the cluster name as a key with an empty value.

Reply:
Ah, good spot. Thanks @rudoi

Reply:
This has been changes by @michaelbeaumont